src/oracle/monitor.js

```diff
@@ -0,0 +1,429 @@
+ */
+const readOracleAddresses = () => {
+    var oracles = readJSONFile(ORACLE_FILE)
+    return oracles
+}
+
+/**
+ * Function to get oracles feed invitations
+ */
+export const getOraclesInvitations = async () => {
+    //get the feeds
+    feeds = agoricNames.reverse
+
+    //for each oracle
+    for (let oracle in oracles) {
+
+        const current = await getCurrent(oracle, fromBoard, { vstorage });
+        const invitations = current.offerToUsedInvitation
+
+        //for each invitation
+        for (let inv in invitations) {
+            let boardId = invitations[inv].value[0].instance.boardId
+            let feed = feeds[boardId].split(" price feed")[0]
+
+            if (!("feeds" in oracles[oracle])) {
+                oracles[oracle]["feeds"] = {}
+            }
+            //add feed
+            oracles[oracle]["feeds"][String(inv)] = feed
+        }
+    }
+}
+
+//var oracleLabels = readOracles();
+var oracles = readOracleAddresses();
+await getOraclesInvitations();
+
+/**
+ * Function to update metrics
+ * @param {*} oracleName oracle name
+ * @param {*} oracle oracle address
+ * @param {*} feed feed name
+ * @param {*} value new feed value
+ * @param {*} id submission id which is a timestamp
+ * @param {*} actualPrice feed actual aggregated price
+ * @param {*} lastRound latest round id for which there was a submission
+ */
+const updateMetrics = (oracleName, oracle, feed, value, id, actualPrice, lastRound) => {
+    //calculate price deviation from actual value
+    let priceDeviation = Math.abs((value - actualPrice) / actualPrice) * 100
+
+    oracleSubmission.labels(oracleName, oracle, feed).set(value)
+    oracleObservation.labels(oracleName, oracle, feed).set(id)
+    oracleLastRound.labels(oracleName, oracle, feed).set(lastRound)
+    oracleDeviation.labels(oracleName, oracle, feed).set(priceDeviation)
+    actualPriceGauge.labels(feed).set(actualPrice)
+}
+
+/**
+ * Function to update balance metrics
+ * @param {*} oracleName oracle name
+ * @param {*} oracle oracle address
+ * @param {*} brand brand
+ * @param {*} value balance value to set
+ */
+const updateBalanceMetrics = (oracleName, oracle, brand, value) => {
+    oracleBalance.labels(oracleName, oracle, brand).set(value)
+}
+
+/**
+ * Function to query price for feed
+ * @param {*} feed feed like 'BRAND_IN-BRAND_OUT'
+ * @returns the price of the feed
+ */
+const queryPrice = async (feed) => {
+    try {
+        const capDataStr = await vstorage.readLatest(
+            `published.priceFeed.${feed}_price_feed`,
+        );
+
+        //parse the value
+        var capData = JSON.parse(JSON.parse(capDataStr).value)
+        capData = JSON.parse(capData.values[0])
+        //replace any extra characters
+        capData = JSON.parse(capData.body.replaceAll("\\", ""))
+
+        //get the latest price by dividing amountOut by amountIn
+        var latestPrice = Number(capData.amountOut.value.digits) / Number(capData.amountIn.value.digits)
```

Comment:
```suggestion
        let latestPrice = Number(capData.amountOut.value.digits) / Number(capData.amountIn.value.digits)
```

Reply:
Also, doesn't fit 80 char limit

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/d2ad409320814196712fd443cdb87a7d625d215b